sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -988,6 +988,8 @@ primaryExpression
     | name=(PERCENTILE_CONT | PERCENTILE_DISC) LEFT_PAREN percentage=valueExpression RIGHT_PAREN
         WITHIN GROUP LEFT_PAREN ORDER BY sortItem RIGHT_PAREN
         (FILTER LEFT_PAREN WHERE where=booleanExpression RIGHT_PAREN)? ( OVER windowSpec)?     #percentile
+    | MODE LEFT_PAREN RIGHT_PAREN WITHIN GROUP LEFT_PAREN ORDER BY sortItem RIGHT_PAREN
```

Comment:
is `mode` the only function that supports WITH GROUP? It looks messy to add one parser rule for each function to add WITH GROUP support.

Reply:
PERCENTILE_CONT and PERCENTILE_DISC also supports WITH GROUP. But mode no need `percentage=valueExpression`.